site/content/docs/main/restore-reference.md

```diff
@@ -36,57 +22,118 @@ Available Commands:
   logs        Get restore logs
 ```
 
-## What happens to NodePorts when restoring Services
+## Detailed Restore workflow
 
-**Auto assigned** NodePorts **deleted** by default and Services get new **auto assigned** nodePorts after restore.
+The following is an overview of Velero's restore process that starts after you run `velero restore create`.
 
-**Explicitly specified** NodePorts auto detected using **`last-applied-config`** annotation and **preserved** after restore.  NodePorts can be explicitly specified as .spec.ports[*].nodePort field on Service definition.
+1. The Velero client makes a call to the Kubernetes API server to create a [`Restore`](api-types/restore.md) object.
 
-#### Always Preserve NodePorts
+1. The `RestoreController` notices the new Restore object and performs validation.
 
-It is not always possible to set nodePorts explicitly on some big clusters because of operation complexity. Official Kubernetes documents states that preventing port collisions is responsibility of the user when explicitly specifying nodePorts:
+1. The `RestoreController` fetches basic information about the backup being restored, like the [BackupStorageLocation](locations.md) (BSL). It also a tarball of the cluster resources in the backup, any volumes that will be restored using Restic, and any volume snapshots to be restored.
 
-```
-If you want a specific port number, you can specify a value in the `nodePort` field. The control plane will either allocate you that port or report that the API transaction failed. This means that you need to take care of possible port collisions yourself. You also have to use a valid port number, one that's inside the range configured for NodePort use.
+1. The `RestoreController` then extracts the tarball of backup cluster resources to the /tmp folder and performs some pre-processing on the resources, including
+
+    * Sorting the resources to help Velero decide the [restore order](#resource-restore-order) to use.
+
+    * Attempting to discover the resources by their Kubernetes [Group Version Resource (GVR)](https://kubernetes.io/docs/reference/using-api/api-concepts/). If a resource is not discoverable, Velero will exclude it from the restore. See more about how [Velero backs up API versions](#backed-up-api-versions)
+
+    * Applying any configured [resource filters](resource-filtering.md).
+
+    * Remapping target namespace according to specified [`--namespace-mappings`](#restoring-into-a-different-namespace) restore option.
+
+1. The `RestoreController` begins restoring the eligible resources one at a time. Valero unmarshalls the current resource into a Kubernetes resource object. Depending on the type of resource and restore options you specified, Velero will make the following modifications to the resource or preparations to the target cluster before attempting to create the resource,
+
```

Comment:
What does "unmarshalls" mean in this context?

Reply:
so i debated a good while about using this word. from my understanding it refers to the go concept/command of marshalling/unmarshalling for converting data from how we have it stored in a velero backup to the format it needs to be in to be understood as a kubernetes object so we can validate it/apply it to the target cluster. 

Probably more universally understood use something like "extract" though :)

Reply:
Yes, "unmarshall" is a Golang concept to convert something like a json to Kubernetes object.
Therefore, "extract" is accurate and better.